template.yaml

```diff
@@ -321,73 +321,86 @@ Resources:
             Action: sts:AssumeRole
       Path: /
       Policies:
-        - PolicyName: ScopusBucketAccess
+        - PolicyName: NvaScopusBucketAccess
           PolicyDocument:
             Version: '2012-10-17'
             Statement:
               - Effect: Allow
                 Action:
-                  - s3:*
-                Resources: '*'
+                  - s3:GetObject
+                Resources:
+                  - !Sub 'arn:aws:s3:::${ScopusXmlBucket}-${AWS::AccountId}/*'
 
-  DefaultLambdaPermissions:
-    Type: AWS::IAM::Policy
+  ScopusUnzipperFunctionRole:
+    Type: AWS::IAM::Role
     Properties:
-      PolicyName: !Sub DefaultLambdaPermissions-${AWS::StackName}
-      Roles:
-        - !Ref ReadFromScopusBucketsRole
-      PolicyDocument:
+      AssumeRolePolicyDocument:
         Version: '2012-10-17'
         Statement:
           - Effect: Allow
-            Action:
-              - logs:CreateLogGroup
-              - logs:CreateLogStream
-              - logs:PutLogEvents
-              - lambda:InvokeFunction
-            Resource: "*"
-
+            Principal:
+              Service: lambda.amazonaws.com
+            Action: sts:AssumeRole
+      Path: /
+      Policies:
+        - PolicyName: ScopusUnzipperBucketAccess
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action:
+                  - s3:PutObject
+                Resources:
+                  - !Sub 'arn:aws:s3:::${ScopusZipBucket}-${AWS::AccountId}/*'
```

Comment:
This is a reference to the bucket's name  (i.e.,  !GetAtt ScopuZipBucket.Name). Adding the account id is not necessary

Reply:
It is best to use the parameters as little as possible and use the !Ref and !GetAtt functions using the actual AWS entities. This gives more flexibility